src/discussions/comments/CommentsView.jsx

```diff
@@ -33,6 +35,9 @@ function CommentsView({ intl }) {
       clearTimeout(markReadTimer);
     };
   }, [postId]);
+  useEffect(() => {
+    dispatch(fetchThreadComments(postId, { page: currentPage }));
+  }, [currentPage]);
```

Comment:
This should depend on both the currentPage and postId, otherwise the same posts will be displayed as you change posts.

Reply:
Unfortunately, even with that change, we see old comments as we switch from one post to another. I think we will need to have pagination on a per-post level. 
i.e. sturcture pagination as follows

```
state.pagination = {
   [postId]: {
       currentPage: number,
       totalPages: numner,
       ids: []
  }
}
```